pkg/resources/notification_integration.go

```diff
@@ -141,180 +160,182 @@ func NotificationIntegration() *schema.Resource {
 // CreateNotificationIntegration implements schema.CreateFunc.
 func CreateNotificationIntegration(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
-	name := d.Get("name").(string)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
 
-	stmt := snowflake.NewNotificationIntegrationBuilder(name).Create()
+	name := d.Get("name").(string)
+	id := sdk.NewAccountObjectIdentifier(name)
+	enabled := d.Get("enabled").(bool)
 
-	// Set required fields
-	stmt.SetString(`TYPE`, d.Get("type").(string))
-	stmt.SetBool(`ENABLED`, d.Get("enabled").(bool))
+	createRequest := sdk.NewCreateNotificationIntegrationRequest(id, enabled)
 
-	// Set optional fields
 	if v, ok := d.GetOk("comment"); ok {
-		stmt.SetString(`COMMENT`, v.(string))
-	}
-	if v, ok := d.GetOk("direction"); ok {
-		stmt.SetString(`DIRECTION`, v.(string))
-	}
-	if v, ok := d.GetOk("azure_tenant_id"); ok {
-		stmt.SetString(`AZURE_TENANT_ID`, v.(string))
-	}
-	if v, ok := d.GetOk("notification_provider"); ok {
-		stmt.SetString(`NOTIFICATION_PROVIDER`, v.(string))
-	}
-	if v, ok := d.GetOk("azure_storage_queue_primary_uri"); ok {
-		stmt.SetString(`AZURE_STORAGE_QUEUE_PRIMARY_URI`, v.(string))
+		createRequest.WithComment(sdk.String(v.(string)))
 	}
-	if v, ok := d.GetOk("azure_tenant_id"); ok {
-		stmt.SetString(`AZURE_TENANT_ID`, v.(string))
-	}
-	if v, ok := d.GetOk("aws_sqs_arn"); ok {
-		stmt.SetString(`AWS_SQS_ARN`, v.(string))
-	}
-	if v, ok := d.GetOk("aws_sqs_role_arn"); ok {
-		stmt.SetString(`AWS_SQS_ROLE_ARN`, v.(string))
-	}
-	if v, ok := d.GetOk("aws_sns_topic_arn"); ok {
-		stmt.SetString(`AWS_SNS_TOPIC_ARN`, v.(string))
-	}
-	if v, ok := d.GetOk("aws_sns_role_arn"); ok {
-		stmt.SetString(`AWS_SNS_ROLE_ARN`, v.(string))
-	}
-	if v, ok := d.GetOk("gcp_pubsub_subscription_name"); ok {
-		stmt.SetString(`GCP_PUBSUB_SUBSCRIPTION_NAME`, v.(string))
-	}
-	if v, ok := d.GetOk("gcp_pubsub_topic_name"); ok {
-		stmt.SetString(`GCP_PUBSUB_TOPIC_NAME`, v.(string))
+
+	notificationProvider := strings.ToUpper(d.Get("notification_provider").(string))
+	switch notificationProvider {
+	case "AWS_SNS":
+		topic, ok := d.GetOk("aws_sns_topic_arn")
+		if !ok {
+			return fmt.Errorf("if you use AWS_SNS provider you must specify an aws_sns_topic_arn")
+		}
+		role, ok := d.GetOk("aws_sns_role_arn")
+		if !ok {
+			return fmt.Errorf("if you use AWS_SNS provider you must specify an aws_sns_role_arn")
+		}
+		createRequest.WithPushNotificationParams(
+			sdk.NewPushNotificationParamsRequest().WithAmazonPushParams(sdk.NewAmazonPushParamsRequest(topic.(string), role.(string))),
+		)
+	case "GCP_PUBSUB":
+		if v, ok := d.GetOk("gcp_pubsub_subscription_name"); ok {
```

Comment:
maybe this param and `gcp_pubsub_topic_name` should be conflicting fields? Now you can specify both and integration will created with `gcp_pubsub_topic_name` only because it's checked later

Reply:
Good idea. I will add it in the following PR, great success?